Oauth

ShreyamsJain opened the pull request:
Tested manually. Facebook login is working. Please review changes

ShreyamsJain commented:
Updated the environment variables(facebook tokens) in Codeship.

ShreyamsJain commented:
Resolved the conflicts. Ready for review

erinmorrissey commented:
@ShreyamsJain I'm trying to test the Facebook login — when I click "Sign In with Facebook", I get redirected to this FB error page: https://www.facebook.com/v2.6/dialog/oauth?client_id&display=popup&redirect_uri=http%3A%2F%2Flocalhost%3A3030%2Fusers%2Fauth%2Ffacebook%2Fcallback&response_type=code&scope=email&state=9c905c8bdb4b36d03f2bd7bcffdb20fc0086c1ee2af3f71c

Server log gives this:
`Started GET "/users/auth/facebook" for 10.0.2.2 at 2016-09-08 22:26:07 +0000
Cannot render console from 10.0.2.2! Allowed networks: 127.0.0.1, ::1, 127.0.0.0/127.255.255.255
I, [2016-09-08T22:26:07.100748 #4182]  INFO -- omniauth: (facebook) Request phase initiated.`

I'm still googling to see what it could be, just thought I'd bring it up in case you happened to encounter it previously?

ShreyamsJain commented:
@erinmorrissey I was facing this issue yesterday, i just dont remember how i fixed it. When you are redirected to the facebook page, are you getting the error "invalid app id"?. Meanwhile i will search and let you know if how i fixed this error.

lasley commented:
only the secret should go in environment vars. You'll still need the `FACEBOOK_APP_ID` var added to your omniauth middleware in the same way you did before the env vars

ShreyamsJain commented:
Made the necessary changes(Added APP_ID to devise.rb) and resolved the conflicts.

erniejamison commented:
Might be working but I'm getting this error from facebook:

App Not Setup: This app is still in development mode, and you don't have access to it. Switch to a registered test user or ask an app admin for permissions.

ShreyamsJain commented:
Sorry for the late response @erniejamison . I checked in facebook settings and there is an option to add developers and testers. If you could tell me your username on Facebook, i can add you as a developer. I suppose, you have to be in my friend list to be added as a developer. Send me a request. My name is Shreyams Jain on fb.

lasley commented:
@ShreyamsJain - The app should be in production mode for our production server. This should not require whitelist

ShreyamsJain commented:
@lasley Ok. There was an option to add developers and testers, so i thought probably the app is in development mode and requires to grant access. There was another option to make the app public and i just turned it on. Maybe it should work now for Ernie.

lasley commented:
@ShreyamsJain - The implementation looks good, but we will need to fix the build errors + add tests
- https://github.com/omniauth/omniauth/wiki/Integration-Testing
- https://gist.github.com/kinopyo/1338738
- https://gist.github.com/spyou/1200365

ShreyamsJain commented:
@erniejamison @erinmorrissey  The files i have been working with are sessions_controller_spec.rb, rails_helper.spec, factories/omniauth.rb. Please ignore support/omniauth_macros.rb

lasley commented on app/models/pieces/untitled.Appraisals:
```diff
@@ -0,0 +1,16 @@
+class King < Piece
```
What is this file?

lasley commented on .rubocop.yml:
```diff
@@ -7,3 +7,6 @@ AllCops:
 Metrics/LineLength:
   Max: 100
+
+Style/BracesAroundHashParameters:
```
Why this change?

lasley commented on app/views/devise/registrations/new.html.erb:
```diff
@@ -26,4 +26,5 @@
   </div>
 <% end %>
 
+
```
Why this change?

lasley commented on coverage/.last_run.json:
```diff
@@ -1,5 +1,5 @@
 {
```
Remove this file

lasley replied:
(this whole coverage folder needs to be deleted)

lasley commented on coverage/.resultset.json:
```diff
@@ -1,7 +1,1552 @@
 {
```
Remove this file

lasley commented on coverage/assets/0.10.0/application.css:
```diff
@@ -0,0 +1,799 @@
+/* -----------------------------------------------------------------------
```
Remove this file

lasley commented:
Hey @ShreyamsJain - we have a conflict on the controller spec due to my #38. Will need to resolve please.

ShreyamsJain commented:
Hi @lasley , i updated my mac os to Sierra and after that i am unable to launch vagrant. Working on fixing that issue. Will resolve the above conflicts as soon as i get vagrant back working.

ShreyamsJain commented:
@lasley This conflicts are resolved.

lasley commented on .gitignore:
```diff
@@ -12,6 +12,9 @@
 /tmp
 
+# Ignore application configuration
+/config/application.yml
```
Why?

ShreyamsJain commented:
Hi @lasley . This has been removed from gitignore

erinmorrissey commented:
I was going to approve and merge this, but the codecov scores are preventing it, at reasonable diff scores for this particular PR, I would say. I saw in codecov.yml where we manually set a coverage range of 80-100, but don't see if there's a GitHub setting that would cause a 90% score to cause a failing check...?

lasley commented:
You can alway override the checks with admin rights. The issue is the diff, not the total. 10% diff from target will pretty much never be considered acceptable. There's a setting in the Codecov YML but I can't remember it TBH

![image](https://cloud.githubusercontent.com/assets/7397578/19385362/f279e54e-91c2-11e6-9ff9-2d8bc4d34c6e.png)
